File: src/main/java/nu/peg/slack/pt/api/slack/model/ActionButtonStyle.java
package nu.peg.slack.pt.api.slack.model;

import com.fasterxml.jackson.annotation.JsonValue;
import com.google.gson.annotations.SerializedName;

public enum ActionButtonStyle {

    @SerializedName("default")
    Default,

    @SerializedName("primary")
    Primary,

    @SerializedName("danger")
    Danger;

    @JsonValue
    public String getSerialized() {
        return this.name().toLowerCase();
    }
}

File: src/main/kotlin/nu/peg/slack/pt/api/slack/model/ActionButtonStyle.kt
package nu.peg.slack.pt.api.slack.model

import com.google.gson.annotations.SerializedName

import com.fasterxml.jackson.annotation.JsonValue

enum class ActionButtonStyle {

    @SerializedName("default")
    Default,

    @SerializedName("primary")
    Primary,

    @SerializedName("danger")
    Danger;

    val serialized: String
        @JsonValue
        get() = this.name.toLowerCase()
}
